bliss/conf/base_config.yaml

```diff
@@ -142,6 +143,7 @@ training:
     seed: 42
     weight_save_path: null
     use_cached_simulator: false
+    use_dc2: false
```

Comment:
I think perhaps we should delete both `use_cached_simulator` and `use_dc2`.
Instead, we'd have a new field called `data_source`, which would be set to either a Simulator, CacheSimulator, or DC2.
e.g., `data_source: $simulator`

Reply:
or, for example, `data_source: ${surveys.dc2}`

Reply:
@zhixiangteoh @XinyueLi1012 Would that make sense? (It seems like we shouldn't have to add another binary flag here anytime we add a data source.)

Reply:
I still think we should reuse `cfg.cached_simulator` and make `DC2` is-a `CachedSimulatedDataset`. Because DC2 dataset is essentially a disk dataset which is what `CachedSimulatedDataset` is—despite the slightly misleading name it actually does not do any simulation itself, since the simulation is done at generate step (which we skip for DC2). So we just set the `cfg.cached_simulator.cached_data_path` appropriately and reuse this dataset class as input to encoder, obviating the need for an additional `data_source` conditional parameter, just having `cfg.cached_simulator.cached_data_path` be the only thing that varies.

Reply:
On another note, `use_cached_simulator` is a boolean that allows us to still simulate data on the fly, if we want to. However if we are moving away completely from that, then we can remove the parameter?

Reply:
We can still simulate on the fly `data_source: ${simulator}`. Not that we'd necessarily want to at this point because of how slow the simulator is. But I'm optimistic we can make it faster in the future (e.g., with jax-galsim or something like it we develop ourselves)